File: experiments-.net/Day04.Planning/Evaluator.cs
using AiAgent.Shared;

namespace Day04.Planning;

public class Evaluator
{
    public (bool needReplan, string? reason) Evaluate(AgentState state, string? observation)
    {
        if (!string.IsNullOrEmpty(observation))
        {
            return (true, observation);
        }
        return (false, null);
    }
}

File: experiments-py/week02/day15-agent-rag/agent/evaluator.py
from models import Evaluation


class Evaluator:
    # 用途：判断结果质量
    def evaluate(self, state, observation):
        if observation.issues:
            return Evaluation(need_replan=True, reason=observation.issues)

        return Evaluation(success=True, need_replan=False)
